odoo/addons/base/models/ir_attachment.py

```diff
@@ -50,7 +50,7 @@ def _compute_res_name(self):
     def _compute_res_model_name(self):
         for record in self:
             if record.res_model:
-                model = self.env['ir.model'].search([('model', '=', record.res_model)], limit=1)
+                model = self.env['ir.model'].sudo().search([('model', '=', record.res_model)], limit=1)
```

Comment:
It isn't something to change now but with  PR https://github.com/odoo/odoo/pull/28563, we will be able to remove this sudo.